ruddock/modules/rotation/routes.py

```diff
@@ -0,0 +1,43 @@
+import json
+import flask
+import io
+
+from ruddock.resources import Permissions
+from ruddock.decorators import login_required
+from ruddock.modules.rotation import blueprint, helpers
+
+
+@blueprint.route('/')
+@login_required(Permissions.ROTATION)
+def show_portal():
+  dinner_list = helpers.get_dinners()
+  return flask.render_template('portal.html',
+    dinner_list=dinner_list)
+
+@blueprint.route('/dinner/<dinner_id>')
+@login_required(Permissions.ROTATION)
+def show_dinner_list(dinner_id):
+  prefrosh_list = helpers.get_prefrosh_by_dinner(dinner_id)
+  return flask.render_template('directory.html',
+    prefrosh_list=prefrosh_list, dinner_id=dinner_id)
+
+@blueprint.route('/images/<prefrosh_id>')
+@login_required(Permissions.ROTATION)
+def serve_image(prefrosh_id):
+  img_contents = helpers.get_image_contents(prefrosh_id)
+  return flask.Response(io.BytesIO(img_contents), '200 OK', None, 'image/jpeg', 'image/jpeg')
+
+@blueprint.route('/prefrosh/<prefrosh_id>')
+@login_required(Permissions.ROTATION)
+def show_prefrosh(prefrosh_id):
+  prefrosh = helpers.get_prefrosh(prefrosh_id)
+  pref = ' \"' + prefrosh['preferred_name'] + '\" ' if prefrosh['preferred_name'] else ' '
```

Comment:
Backslashes? Are you trying to go for something that looks like this `Daniel "DK" Kong`? This is a minor nit, but the preferred name isn't supposed to be an infix ype thing, its more if someone legitimately goes by another name, so something like `Ching-Yun (Chloe) Hsu` would look better imo.

Also this code is fairly hacky and hard to read. Something like this might be better:
```python
name_parts = []
name_parts.append(prefrosh["first_name"])
if prefrosh["preferred_name"] is not None:
  name_parts.append("({})".format(prefrosh["preferred_name"]))
name_parts.append(prefrosh["last_name"])
full_name = " ".join(name_parts)
```